lib/types.nix

```diff
@@ -219,8 +219,7 @@ rec {
 
     path = mkOptionType {
       name = "path";
-      # Hacky: there is no ‘isPath’ primop.
-      check = x: builtins.substring 0 1 (toString x) == "/";
+      check = x: isString x || isPath x;
```

Comment:
I'm wondering whether this was intended to be `isString x && isPath x`. As-is, wouldn't some nonsense value like "$%^&*" be accepted, as long as it is a string?

Reply:
> As-is, wouldn't some nonsense value like "$%^&*" be accepted, as long as it is a string?

Yeah, that seems to be the case. Is that intended?

Reply:
I don't see why arbitrary strings should be accepted as a path.

Reply:
Yes, this needs to allow (string or path type) paths.

Reply:
It would need to be `(isString x && is absolute path) || isPath x`. Will try to fix that.